server/messages.go

```diff
@@ -131,7 +129,10 @@ func processMessages(conn net.Conn, c chan *packetInfo, t *Tracker) {
 
 			if b.Len() >= numReadBytes {
 				msg := make([]byte, numReadBytes)
-				b.Read(msg)
+				if _, err := b.Read(msg); err != nil {
+					log.Warnf(logCommunication+"Error reading from message buffer: %s\n", err)
+					return
+				}
```

Comment:
@zquestz I don't think I asked about this before but I inserted a return here. Was there a specific reason for allowing read errors here?

Reply:
I think it was just an oversight. This should be fine. There is already a length check so this read should always succeed!